hivelvet-backend/app/src/Actions/Presets/Add.php

```diff
@@ -0,0 +1,185 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * Hivelvet open source platform - https://riadvice.tn/
+ *
+ * Copyright (c) 2022 RIADVICE SUARL and by respective authors (see below).
+ *
+ * This program is free software; you can redistribute it and/or modify it under the
+ * terms of the GNU Lesser General Public License as published by the Free Software
+ * Foundation; either version 3.0 of the License, or (at your option) any later
+ * version.
+ *
+ * Hivelvet is distributed in the hope that it will be useful, but WITHOUT ANY
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+ * PARTICULAR PURPOSE. See the GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License along
+ * with Hivelvet; if not, see <http://www.gnu.org/licenses/>.
+ */
+
+namespace Actions\Presets;
+
+use Actions\Base as BaseAction;
+use Base;
+use Enum\ResponseCode;
+use Models\Preset;
+use Models\PresetCategory;
+use Models\PresetSetting;
+use Models\SubCategory;
+use Models\SubCategoryPreset;
+
+/**
+ * Class Add.
+ */
+class Add extends BaseAction
+{
+    /**
+     * @param Base  $f3
+     * @param array $params
+     */
+    public function save($f3, $params): void
+    {
+        $body = $this->getDecodedBody();
+        $this->logger->info(' body', $body);
+
+        $form = $body['data'];
+
+        $preset       = new Preset();
+        $preset->name = $form['name'];
+
+        try {
+            $preset->save();
+            $presetData = [
+                'name'       => $preset->name,
+                'categories' => [],
+            ];
+        } catch (\Exception $e) {
+            $message = '  could not be added';
+            $this->logger->error('  could not be added', ['error' => $e->getMessage()]);
+            $this->renderJson(['message' => $e->getMessage()], ResponseCode::HTTP_INTERNAL_SERVER_ERROR);
+
+            return;
+        }
+
+        $presetsett = new PresetSetting();
+
+        $data                = [];
+        $classes             = get_declared_classes();
+        $autoloaderClassName = '';
+        foreach ($classes as $className) {
+            if (str_starts_with($className, 'ComposerAutoloaderInit')) {
+                $autoloaderClassName = $className;
+
+                break;
+            }
+        }
+        $classLoader = $autoloaderClassName::getLoader();
+        $classMap    = $classLoader->getClassMap();
+
+        $categories = preg_filter('/^Enum\\\Presets\\\[A-Z a-z]*/', '$0', array_keys($classMap));
```

Comment:
Looks like a Titan work :rocket: !! Well done. However can you help simple human understand it by adding comment lines?